draft-ietf-jsonpath-base.md

```diff
@@ -569,206 +581,226 @@ of node.
-child = %x22 *double-quoted %x22 / ; "string"
-        "'" *single-quoted "'"   ; 'string'
-
-double-quoted = dq-unescaped /
-          escape (
-              %x22 /         ; "    quotation mark  U+0022
-              "/" /          ; /    solidus         U+002F
-              "\" /          ; \    reverse solidus U+005C
-              "b" /          ; b    backspace       U+0008
-              "f" /          ; f    form feed       U+000C
-              "n" /          ; n    line feed       U+000A
-              "r" /          ; r    carriage return U+000D
-              "t" /          ; t    tab             U+0009
-              "u" 4HEXDIG )  ; uXXXX                U+XXXX
-
+index-selector      = "[" (quoted-member-name / element-index) "]"
+~~~~
 
-dq-unescaped = %x20-21 / %x23-5B / %x5D-10FFFF
+Applying the `index-selector` to an object value, a `quoted-member-name` string is required. JSONPath allows it to be enclosed in _single_ or _double_ quotes.
 
-single-quoted = sq-unescaped /
-          escape (
-              "'" /          ; '    apostrophe      U+0027
-              "/" /          ; /    solidus         U+002F
-              "\" /          ; \    reverse solidus U+005C
-              "b" /          ; b    backspace       U+0008
-              "f" /          ; f    form feed       U+000C
-              "n" /          ; n    line feed       U+000A
-              "r" /          ; r    carriage return U+000D
-              "t" /          ; t    tab             U+0009
-              "u" 4HEXDIG )  ; uXXXX                U+XXXX
+~~~~ abnf
+quoted-member-name  =  %x22 *double-quoted %x22 /       ; "string"
+                       %x27 *single-quoted %x27         ; 'string'
+
+double-quoted       = unescaped /
+                      %x27      /                       ; '
+                      ESC %x22  /                       ; \"
+                      ESC escapable
+
+single-quoted       = unescaped /
+                      %x22      /                       ; "
+                      ESC %x27  /                       ; \'
+                      ESC escapable
+
+ESC                 = %x5C                              ; \  backslash
+
+unescaped           = %x20-21 /                         ; s. RFC 8259
+                      %x23-26 /                         ; omit "
+                      %x28-5B /                         ; omit '
+                      %x5D-10FFFF                       ; omit \
+
+escapable           = (
+                          b /         ;  BS backspace U+0008
+                          t /         ;  HT horizontal tab U+0009
+                          n /         ;  LF line feed U+000A
+                          f /         ;  FF form feed U+000C
+                          r /         ;  CR carriage return U+000D
+                          / /         ;  /  slash (solidus)
+                          \ /         ;  \  backslash (reverse solidus)
+                          u 4HEXDIG   ;  uXXXX      U+XXXX
```

Comment:
See https://www.ietf.org/archive/id/draft-bormann-cbor-cddl-freezer-07.html#name-err6527 for how to do this right.  (Again, this should be in the expression language section.)

Reply:
Wouldn't make it sense then to simple copy that into here ?

Reply:
Yes.
(Discuss today.)

Reply:
Well, slight massaging required to support both single-quoted and double-quoted strings.
I think we should leave this to a followup PR.

Reply:
Resolved as per interim (needed a bit more surgery than I had thought, though).